### PR TITLE
phpExtensions.datadog_trace: 0.82.0 -> 0.89.0

### PR DESCRIPTION
**Pull request by drupol**
Thanks @figsoda for the help and for telling me to use the amazing [nix-init](https://github.com/nix-community/nix-init). This is thanks to it that I was able to fix this.

An upstream issue has been opened: https://github.com/DataDog/dd-trace-php/issues/2151 but since the issue was coming from Nix, I closed it.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/php-packages/datadog_trace/default.nix [hunk 1]
```diff
@@ -1,22 +1,51 @@
...
+    };
+  };
+
+  env.NIX_CFLAGS_COMPILE = "-O2";
+
+  nativeBuildInputs = [
+    rustPlatform.bindgenHook
+    rustPlatform.cargoSetupHook
+    rustPlatform.rust.cargo
+    rustPlatform.rust.rustc
```
Comment (figsoda) [inline pkgs/development/php-packages/datadog_trace/default.nix:36]: `rustPlatform.rust.*` has been deprecated, this has been fixed in nix-init 0.2.4 / nixpkgs master

```suggestion
    cargo
    rustc
```

Comment (drupol) [reply pkgs/development/php-packages/datadog_trace/default.nix]: Fixed.

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extenxions.datadog_trace
@GrahamcOfBorg build php82Extenxions.datadog_trace
@GrahamcOfBorg build php83Extenxions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace
@GrahamcOfBorg build php82Extensions.datadog_trace
@GrahamcOfBorg build php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace php82Extensions.datadog_trace php83Extensions.datadog_trace

File: pkgs/build-support/build-pecl.nix [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{ stdenv, lib, php, autoreconfHook, fetchurl, re2c, nix-update-script }:
+{ stdenv, lib, php, autoreconfHook, fetchurl, re2c, nix-update-script }@imports:
```
Comment (SuperSandro2000) [inline pkgs/build-support/build-pecl.nix:1]:
```suggestion
{ stdenv, lib, php, autoreconfHook, fetchurl, re2c, nix-update-script }@args:
```
that is the standard variable we use for this. Also unless we are using ... @args does not much.

File: pkgs/development/php-packages/datadog_trace/default.nix [hunk 3]
```diff
@@ -1,22 +1,56 @@
...
+, php
+, cargo
+, rustc
+, fetchFromGitHub
+, rustPlatform
+, curl
+, pcre2
+, libiconv
+, darwin
+}@attrs:
```
Comment (SuperSandro2000) [inline pkgs/development/php-packages/datadog_trace/default.nix:12]: same here

File: pkgs/development/php-packages/datadog_trace/default.nix [hunk 4]
```diff
@@ -1,22 +1,56 @@
...
 
-buildPecl rec {
-  pname = "ddtrace";
-  version = "0.82.0";
+php.buildPecl rec {
+  stdenv = if attrs.stdenv.isDarwin then darwin.apple_sdk_11_0.stdenv else attrs.stdenv;
```
Comment (SuperSandro2000) [inline pkgs/development/php-packages/datadog_trace/default.nix:15]: That is not necessary. darwin.apple_sdk_11_0.stdenv is stdenv for linux.

File: pkgs/development/php-packages/datadog_trace/default.nix [hunk 5]
```diff
@@ -1,22 +1,56 @@
...
+  pname = "datadog_trace";
+  version = "0.89.0";
 
   src = fetchFromGitHub {
     owner = "DataDog";
     repo = "dd-trace-php";
     rev = version;
-    sha256 = "sha256-QTqZRHh57mRkg0HT9qQS13emGobB0IRqM+mdImAPgtE=";
+    hash = "sha256-wTGQV80XQsBdmTQ+xaBKtFwLO3S+//9Yli9aReXDlLA=";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/php-packages/datadog_trace/default.nix:25]:
```suggestion
    fetchSubmodules = true;
    hash = "sha256-wTGQV80XQsBdmTQ+xaBKtFwLO3S+//9Yli9aReXDlLA=";
```

File: pkgs/development/php-packages/datadog_trace/default.nix [hunk 6]
```diff
@@ -1,22 +1,56 @@
...
+    cargo
+    rustc
+    rustPlatform.bindgenHook
+    rustPlatform.cargoSetupHook
+  ];
+
+  buildInputs = [ curl pcre2 libiconv ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk_11_0.frameworks.CoreFoundation
+    darwin.apple_sdk_11_0.frameworks.Security
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/php-packages/datadog_trace/default.nix:47]:
```suggestion
  buildInputs = [ curl pcre2 ]
  ++ lib.optionals stdenv.isDarwin [
    darwin.apple_sdk_11_0.frameworks.CoreFoundation
    darwin.apple_sdk_11_0.frameworks.Security
    libiconv
  ];
```
libiconv is null on linux

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace php82Extensions.datadog_trace php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace php82Extensions.datadog_trace php83Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace php82Extensions.datadog_trace php83Extensions.datadog_trace

File: pkgs/development/php-packages/datadog_trace/default.nix (hunk 4 above)
Comment (drupol) [reply pkgs/development/php-packages/datadog_trace/default.nix]: When I do that, I get the following:

```
❯ nix build .#php.extensions.datadog_trace -L
warning: Git tree '/home/devlin/Code/NixOS/nixpkgs' is dirty
error: Package ‘apple-framework-CoreFoundation-11.0.0’ in /nix/store/pnjm969l5vsskflcydqik3ahbch3inhh-source/pkgs/os-specific/darwin/apple-sdk-11.0/apple_sdk.nix:77 is not available on the requested hostPlatform:
         hostPlatform.config = "x86_64-unknown-linux-gnu"
         package.meta.platforms = [
           "x86_64-darwin"
           "i686-darwin"
           "aarch64-darwin"
           "armv7a-darwin"
         ]
         package.meta.badPlatforms = [ ]
       , refusing to evaluate.

       a) To temporarily allow packages that are unsupported for this system, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1

        Note: For `nix shell`, `nix build`, `nix develop` or any other Nix 2.4+
        (Flake) command, `--impure` must be passed in order to read this
        environment variable.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnsupportedSystem = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnsupportedSystem = true; }
       to ~/.config/nixpkgs/config.nix.
(use '--show-trace' to show detailed location information)
~/C/N/nixpkgs > php/extensions/bump-datadog-trace +1 -1 [!] ✘ 
```

File: pkgs/build-support/build-pecl.nix (hunk 2 above)
Comment (drupol) [reply pkgs/build-support/build-pecl.nix:1]: The `build-pecl.nix` is a binary lambda function. There's a first set of parameters using `@imports`, and the second one using `@args` already.

Should I change something there?

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace

**Comment by drupol** [comment]
@GrahamcOfBorg build php81Extensions.datadog_trace

**Comment by drupol** [comment]
Follow up PR: https://github.com/NixOS/nixpkgs/pull/242655

**Pull request merged**